h2o-r/h2o-package/R/automl.R

```diff
@@ -249,7 +249,7 @@ h2o.automl <- function(x, y, training_frame,
   is_progress <- isTRUE(as.logical(.h2o.is_progress()))
   h2o.no_progress()
   leaderboard <- tryCatch(
-    as.h2o(leaderboard),
+    as.h2o(leaderboard, paste(build_control$project_name, "leaderboard", sep=".")),
```

Comment:
please note that I suspect there are several other cases where we may override an object on the backend due to implicit generation of destination_frame when calling `as.h2o` or `as.h2o.data.frame`.
we may want to review those usages

Reply:
Is the idea here to give the leaderboard a unique key?

Reply:
Is the idea here to give the leaderboard frame a unique key per automl project? What was happening before?

Reply:
@seb-h2o should the fix then be in `as.h2o.data.frame` instead of in AutoML?

I am adding @jangorecki to the list of reviewers. @jangorecki wdyt?

Reply:
@michalkurka Better to fix the upstream project, but I cannot see clearly defined issue for `h2o` package behavior. Do we want to `test$Sepal.Length <- 1` to update by reference (and keep origin `id`)?

Reply:
@jangorecki the `test$Sepal.Length <- 1` has no impact on the issue, I just reused Megan's code for the UT, let me remove this statement to avoid confusion.
The exception occured any time there was a Rapids request on the first leaderboard object AFTER the second automl had been run: this is because the local leaderboard reference somehow didn't match the one stored on backend which had been replaced in between by the second leaderboard.

@michalkurka that's what I was suggesting on #2792, basically generate a random key like we do for `as.h2o.H2OFrame`: will submit this here.